### ISSUE TITLE
`start` and `end` for location and zero base

### ISSUE THREAD

**Issue by axetroy** [denoland/deno_lint#290]
before:

```json
{
 "diagnostics": [
    {
     "location": {
        "filename": "/path/to/file/mod.ts",
         "line": 1,
         "col": 0,
      },
     "message": "`debugger` statement is not allowed",
     "code": "no-debugger",
     "line_src": "debugger;",
     "snippet_length": 9,
    },
  ],
 "errors": [],
}
```

after:

```json
{
 "diagnostics": [
    {
	  "file": "/path/to/file/mod.ts",
      "location": {
	  	"start": {
          "line": 0,
           "col": 0
        },
		"end": {
          "line": 0,
          "col": 8
        }
      },
      "message": "`debugger` statement is not allowed",
      "code": "no-debugger",
      "line_src": "debugger;",
      "snippet_length": 9,
    },
  ],
  "errors": [],
}
```

**Comment by bartlomieju** [comment]
Thanks @axetroy I actually intended to implement this, with recent changes to `Location` struct. 

`snippet_length` and `line_src` fields will be removed

**Comment by axetroy** [comment]
Good to hear. Here is the implement of vscode

```ts
/**
 * A range in a text document expressed as (zero-based) start and end positions.
 *
 * If you want to specify a range that contains a line including the line ending
 * character(s) then use an end position denoting the start of the next line.
 * For example:
 * ```ts
 * {
 *     start: { line: 5, character: 23 }
 *     end : { line 6, character : 0 }
 * }
 * ```
 */
export interface Range {
    /**
     * The range's start position
     */
    start: Position;
    /**
     * The range's end position.
     */
    end: Position;
}

/**
 * Position in a text document expressed as zero-based line and character offset.
 * The offsets are based on a UTF-16 string representation. So a string of the form
 * `a𐐀b` the character offset of the character `a` is 0, the character offset of `𐐀`
 * is 1 and the character offset of b is 3 since `𐐀` is represented using two code
 * units in UTF-16.
 *
 * Positions are line end character agnostic. So you can not specify a position that
 * denotes `\r|\n` or `\n|` where `|` represents the character offset.
 */
export interface Position {
    /**
     * Line position in a document (zero-based).
     * If a line number is greater than the number of lines in a document, it defaults back to the number of lines in the document.
     * If a line number is negative, it defaults to 0.
     */
    line: number;
    /**
     * Character offset on a line in a document (zero-based). Assuming that the line is
     * represented as a string, the `character` value represents the gap between the
     * `character` and `character + 1`.
     *
     * If the character value is greater than the line length it defaults back to the
     * line length.
     * If a line number is negative, it defaults to 0.
     */
    character: number;
}
```

**Comment by bartlomieju** [comment]
@axetroy looking at `Diagnostic` interface I think we'll only use `range`, `code` and `message`. Can you fill out the rest of the values in the extension?
